### PR TITLE
[WIP] Replaces moths with another roundstart race: Gehennites

### PR DESCRIPTION
**Pull request by Militaires**
image
image
image

Gehennites are a race of blind, echolocation-using humanoid mutants developed by NT lazily, by suppressing the genes in a strain of war mutants that give them their monstrous strength, unfortunately, they were unable to avoid suppressing the genes responsible for vision, echolocation remains .As a result of the gene manipulation, Gehennites can not be cloned.

This PR removes moths as a roundstart race and puts gehennites in their place. Roundstart, they are given nodrop gene-suppression devices, which take up (likely the neck or head) slot, I'll make sure that the sprite for this device is very easily visible to silicons, so that their human appearance cant be abused for human status. They receive more 20% damage than humans for almost everything

### Gene-liberated Form:
These insane mutants are triple the size of humans, and were used by NT for war efforts, their inability to distinguish between friend and enemy made them useless, and they were put into a weakened form and used for cheap labor.
image
edit: note that their human sprite is a placeholder and ill have the time for a more intricate one if the pr is merged, it's a waste to make a really good sprite for something unlikely to be merged.

If a gehennite is hit by an explosion of certain severity, AND also survives the explosion, there is a minuscule (I think I'll go with 0.5/100) chance that the gene suppression device breaks, and their true form is brought out. Assuming there's a 50/50 chance of surviving the explosion, the final probability would be 0.0025 (it should almost never happen) When this happens, their mind is swapped with a ghost's (to symbolize the personality shift and to make even hunting for this probability useless, you won't get to play one)

In this form, they will be dextrous (tool-using) and VERY strong, IF this PR is merged (lol) I'll also expand on their powers and likely give them a power tree/menu like changelings. Finally, they get infrared vision, that lets them see humans but not terrain, almost the opposite of echolocation (which DOES let you see the reflection of mobs but since it's a single image and not a video you wont be seeing anything moving)

Rapidfire:

Why are you replacing moths and not another species for roundstart?
- There is a set number of roundstart races and I can't add a new species without removing an old one (unless a maintainer says otherwise) Moths were practically added because of a long-dead meme, and offer no significant gameplay variation, you could argue the same for lizards but lizards have been in the code much longer and are deeper seated than moths.

Are they human in the AI's eyes?
- For me: No, they are too modified. This is a policy issue though.

Why are they worse than humans on all accounts?
- Humans first policy. I can negotiate the numbers and whether they get more damage though.

:cl: Basilman and Razharas
add: Moths have been removed as a roundstart race, a new race has been added instead.
code: Reworked inhand offset system.
/:cl:

TO-DO:

- [X] Echolocation
- [X] Gehennite Species
- [X] Gene-liberated Gehennite mob
- [ ] Mob interactions 
- [ ] Infrared

i realize this PR has only a low chance of being merged but hey im gonna try anyways

### CODE REVIEW

**Comment by Militaires** [comment]
inb4 instaclose....

**Comment by RaveRadbury** [comment]
So they can't use tools in general?

**Comment by Militaires** [comment]
They can use tools, I just meant the big form can use tools as well.

**Comment by RaveRadbury** [comment]
So there's a small form not currently pictured in the PR?

**Comment by actioninja** [comment]
hell yeah

File: code/_globalvars/lists/objects.dm [hunk 1]
```diff
@@ -32,6 +32,7 @@ GLOBAL_LIST_EMPTY(meteor_list)				// List of all meteors.
 GLOBAL_LIST_EMPTY(active_jammers)             // List of active radio jammers
 GLOBAL_LIST_EMPTY(ladders)
 GLOBAL_LIST_EMPTY(trophy_cases)
+GLOBAL_LIST_EMPTY(echo_images)
```
Comment (optimumtact) [inline code/_globalvars/lists/objects.dm]: is there a reason to have this as a global list and not just a list on SSoutputs?

File: code/datums/components/echolocation.dm [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	for(var/F in filtered)
+		var/atom/S = F
+		if(GLOB.echo_images[S.type])
+			image_output = mutable_appearance(GLOB.echo_images[S.type].icon, GLOB.echo_images[S.type].icon_state, GLOB.echo_images[S.type].layer, GLOB.echo_images[S.type].plane)
+			image_output.filters = GLOB.echo_images[S.type].filters
+			realign_icon(image_output, S)
+		else
+			image_output = generate_image(S)
+			realign_icon(image_output, S)
+			if(!(SSoutputs.uniques[S.type]))
```
Comment (optimumtact) [inline code/datums/components/echolocation.dm]: what is the point of this compared to the echo images global list?

**Comment by Militaires** [comment]
How echolocation works codewise:

Echolocation component runs a view() in 4 turfs range of parent, then adds the walls, mobs, and objs in that list, after that, if it's an obj, it has an image made of it (via icon and icon state) then that image has a byond outline filter applied, and its icon is set to black, the image is cached with its type as a key so the operation is never done again (then sent to mob), and it's pulled from the cache list via type if the object needs to be echolocated, if it's a wall, it gets a block and erases the borders from it if there's an adjacent turf then sends the image to client images, if it's a MOB or is a UNIQUE, it runs getflaticon and outlines that icon and sends it to mob's client's images. Walls and mobs do not get cached but that is completely optional, it's dynamic like this so walls can be smoothed and human outlines can be seen.

If this proves to be laggy (it's instant on a test server) i can easily make it so walls just return a non-smoothed group of blocks and mobs are all one generic icon (generic human icon, generic monkey etc)

@RaveRadbury nah even the normal form can use tools i just wanted to point it out cause big mobs like queens cant use tools

**Comment by necromanceranne** [comment]
Not sure why you would have the chance to go sicko mode be so unlikely, locked behind permadeath, only to have the person be ghosted. That seems lame to me.

Also there are more moth players than lizard players at this point. I don't see why we need to remove a race from roundstart that is still seeing plenty of use. Not even as a meme, people genuinely like moths.

**Comment by Militaires** [comment]
I'll remove whatever species the maintainers want me to replace this with, as for sicko mode, it's made like that because of the HUMAN FIRST (others dont get advantages) policy, also, if it wasnt like that, people would fish for it

**Comment by necromanceranne** [comment]
By fishing you mean kill yourself near every round from an explosion until you luck out? The chance you suggest are so incredibly low as to be unrealistic to achieve via fishing unless the guy was dedicated to wasting a lot of playtime and rounds. And depending on explosion severity and precautions taken may or may not eventually invite a ban from careless station destruction every round.

Also with a 1.2x damage mod explosions will kill very easily. There is no 50/50, you will probably just straight up die.

File: code/datums/components/echolocation.dm [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		H.display_output(, image_output,, S.loc)
+	for(var/T in turfs)
+		image_output = generate_wall_image(T)
+		image_output.loc = T
+		H.display_output(, image_output,, T)
+
+/datum/component/echolocation/proc/generate_image(atom/input)
+	var/icon/I
+	if(SSoutputs.uniques[input.type])
+		I = getFlatIcon(input)
```
Comment (optimumtact) [inline code/datums/components/echolocation.dm]: can't you just get the mutable appearance of the object rather than calling getflaticon?

Comment (Militaires) [reply code/datums/components/echolocation.dm]: well not for humans and walls and shit since they're made up of a bunch of MAs

**Comment by Militaires** [comment]
@necromanceranne even if people didnt fish for it, there is a humans first policy which means all races MUST be subhuman

File: code/datums/components/echolocation.dm (hunk 3 above)
Comment (optimumtact) [reply code/datums/components/echolocation.dm]: that goes for the one below as well, you're turning the result into a mutable apperance anyway, why not just copy the existing atoms appearance and apply the filter to it and then just make sure it appears on the curse layer

**Comment by Nickkoranda** [comment]
based
looks annoying as hell to play as, but also hilariously fun

**Comment by necromanceranne** [comment]
So you made them extra damage permadeath antag spawners?

**Comment by RaveRadbury** [comment]
They're flash-proof then?

**Comment by HomocidlePancakes** [comment]
Remove lizards instead. Atleast moths can be abused with flashes

**Comment by Militaires** [comment]
i think all blind people are flashproof anyhow

File: code/datums/components/echolocation.dm (hunk 3 above)
Comment (Militaires) [reply code/datums/components/echolocation.dm]: because i need to use icon mapcolor proc which only works on icons sadly

**Comment by Nickkoranda** [comment]
Blinded people can still be flashed (and stunned from it). I believe somebody sent it an issue before, but it was marked as intended.

File: code/datums/components/echolocation.dm (hunk 3 above)
Comment (optimumtact) [reply code/datums/components/echolocation.dm]: fyi for anyone else who wondered this, there's no equivalent of MapColors for MA's and if you set the MA.color it bleeds into the outline

Comment (optimumtact) [reply code/datums/components/echolocation.dm]: probably because it applies after the filter

**Comment by bobbahbrown** [comment]
cool idea, ignore the droolies who want to maintain the status quo we need a shakeup in thinking

**Comment by HomocidlePancakes** [comment]
So how do they interact with loud noises like flashbangs or maybe even explosions?

**Comment by ATH1909** [comment]
I really do love this race... but as a moth main, I can't give this PR anything but a downthumb.

If you can persuade a maintainer to increase the number of roundstart races and allow you to add this race to the game without removing moths, I'll remove my downthumb and give this PR an upthumb and a heart.

**Comment by Shadowflame909** [comment]
You should remove plasmamemes instead. They're not living up to their hard mode race lifestyle, and this would be the perfect replacement.

HARDMODE RACE FOR THE WIN

**Comment by Militaires** [comment]
In the future flashbangs and explosions will be tied in with datum outputs, and they'll see a huge overlay on their screen sort of like sound rings I guess

**Comment by Yenwodyah** [comment]
I don't see the point of these guys. Why would players want to play them?
 - Visually they're just grey humans.
 - The awakened form is so rare that the only time it'll show up is right after this is merged, when players bomb themselves all round 10 times in a row to see it'
 - Outside of the novelty, no one will ever WANT to go into the awakened form, since it ghosts the player.
 - Moths have way more flavor and more unique mechanics than these guys (eating clothing, vegetarianism, wings which can burn off, flash vulnerability)

**Comment by Militaires** [comment]
>Moths have way more flavor and more unique mechanics than these guys (eating clothing, vegetarianism, wings which can burn off, flash vulnerability)

Lord forgive me... but Swagger tighter than a yeast infection. Fly, go hard, like geese, erection. moths are reskinned humans and all of the features you mentioned are puddle-deep and rarely affect the game significantly

**Comment by fluffe9911** [comment]
Atleast if this pr ends up being closed you could turn the infrared and echo location stuff into genetics/round start traits if you wanted so that's neat

**Comment by Yenwodyah** [comment]
Echolocation is cool, but I think it would be better as a quirk, a mutation, or some other thing that doesn't tie it to species. It's on the same level as being blind, deaf, or wheelchair-bound, which are all quirks. It would be silly to make a wheelchair species that's uncloneable and more easily damaged to compensate for the fact that they can use wheelchairs, right?
So aside from the echolocation, I really don't see what makes the species anything besides "humans but grey and weaker". From the way you've described it, with such a small chance of triggering from an already-rare form of damage, the awakened form is going to have a lot smaller effect on the game than any other species - moths use their wings to fly around all the time when the power goes out, lizards can speak in their special language, even felinids get harassed for cargo bounties or with laser pointers at more than a .25% chance. 
At the same time, the fact that the awakened form gets handed off to a ghost makes it feel more like an event antag than a species trait. A rarer xeno egg or space dragon spawn. I think that flash-vulnerable, clothes-eating, flying bug aliens are much more interesting than a blind gray human that spawns an event antag .001% of the time when they die.

**Comment by datorangebottle** [comment]
lol so edgy

**Comment by Militaires** [comment]
no other species has a unique feature like echolocation, the language for lizards: even monkeys have their own language, moths using their wings only works in atmos and you can move at their exact speed if you JUST have a wall next to you or a jetpack. you cant go "OH WELL ASIDE FROM THE MAIN FEATURE OF THIS SPECIES, WHICH MAKES IT INTERESTING, IT'S NOT INTERESTING"

**Comment by datorangebottle** [comment]
> no other species has a unique feature like echolocation, the language for lizards: even monkeys have their own language, moths using their wings only works in atmos and you can move at their exact speed if you JUST have a wall next to you or a jetpack. you cant go "OH WELL ASIDE FROM THE MAIN FEATURE OF THIS SPECIES, WHICH MAKES IT INTERESTING, IT'S NOT INTERESTING"

Echolocation is interesting, but it doesn't justify the antag event bit that's extremely dumb and edgy. They're also visually uninteresting as pale humans.

**Comment by necromanceranne** [comment]
More specifically it feels like you're punishing yourself and even the station as a whole for playing one. Which is neat but why we need that as a roundstart race is beyond me.

**Comment by Militaires** [comment]
>Echolocation is interesting, but it doesn't justify the antag event bit that's extremely dumb and edgy. 

can you explain why it's "dumb and edgy", it's a rare event, rarer than something like space dragon or megafauna getting on the station

>They're also visually uninteresting as pale humans.

Sprites are insignificant at this stage.

**Comment by datorangebottle** [comment]
> can you explain why it's "dumb and edgy", it's a rare event, rarer than something like space dragon or megafauna getting on the station

Because it's this massive pitch-black creature rippling with muscles that was being suppressed the government and is now rebelling against authority or whatever. 

> Sprites are insignificant at this stage.

Sprites are pretty damn significant when presenting your idea to be merged to the server.

**Comment by Militaires** [comment]
Coldsteel the Gehennite. Coldsteel was born with a special power, he was stronger than all of his classmates at the NT fighting academy.

@datorangebottle im not gonna make a really great sprite for something unlikely to be merged bro

**Comment by Yenwodyah** [comment]
Like I said, I think that echolocation is a cool feature. I just don't think that it's game-changing enough to justify removing moths and replacing them with an otherwise-bland grey human species. It would work great as a quirk, a mutation, or some kind of augment though. 
The way that it and permadeath are implemented are also a little harsh. Not even plasmamen, the hardmode species, have permadeath, it's just very hard to clone them, and the way that the difficulty for plasmamen is increased creates interesting gameplay possibilities, like setting up a surgery table under a shower, waiting with an extinguisher for them to pop out of the cloning tube, filtering the air out of a room and filling it with plasma so that only plasmamen can breathe, etc. What are some gameplay possibilities for this species, besides "bomb yourself ten times in a row to spawn a megafauna"?

**Comment by Militaires** [comment]
people making plasmaman cloners almost never happens too you know they're PRETTY MUCH permadeath. i'll let oranges pick what stays and goes out in this pr.

**Comment by datorangebottle** [comment]
> @datorangebottle im not gonna make a really great sprite for something unlikely to be merged bro
Then why bother making the race at all? You could've just made a really cool echolocation feature, which you did.

> Not even plasmamen, the hardmode species, have permadeath, it's just very hard to clone them, and the way that the difficulty for plasmamen is increased creates interesting gameplay possibilities, like setting up a surgery table under a shower, waiting with an extinguisher for them to pop out of the cloning tube, filtering the air out of a room and filling it with plasma so that only plasmamen can breathe, etc.

Actually, I could see that as a way to save this species. Plasmamen are no longer anywhere near the 'hardmode species'. Without their slowdown they are an upside to humans in almost every way and I've argued the semantics many many times. 

Make this the hardmode species. Make the sprites less edgy but more unique. Remove the really, really bad gene-released form.  Bam. Echolocation and permadeath sound pretty hardmode to me.

**Comment by necromanceranne** [comment]
You might win more people over by taking out the damage mods. The core mechanic alone might be groovy enough for some to accept it.

**Comment by Militaires** [comment]
i'll consider this

**Comment by Militaires** [comment]
@datorangebottle im not gonna make a really great sprite for something unlikely to be merged bro
sir please read the second half of that post too

**Comment by datorangebottle** [comment]
> sir please read the second half of that post too

I expect a demon king to be edgy, not a roundstart race.

**Comment by Militaires** [comment]
:ok: im hearin ya

**Comment by PKPenguin321** [comment]
So is the idea to just sit in a corner with a medibot hitting yourself with IEDs or

**Comment by actioninja** [comment]
> So is the idea to just sit in a corner with a medibot hitting yourself with IEDs or

you could at least read the whole pr before complaining

**Comment by ATH1909** [comment]
If we do "have to" remove a race to add gehennites (which we don't; the "cap" is completely arbitrary and not there due to code issues (as stated by oranges in Discord) and we could very easily just declare this race to be the "replacement" for flypeople (who have already been removed with no replacement)), why not remove ethereals instead of moths?

When was the last time you saw an ethereal? They're about as rare now as flypeople were just before they were removed. In comparison, you almost always see (at least) one or two moths each round on Bagil. This also ignores that moths have more interesting unique mechanics (wings, eating clothes, and flash-vulnerability) than ethereals do (taking damage when too hungry, eating electricity, and dealing burn damage with punches (a feature so underused/undernoticed, by the way, that a bug that made punch damage dealt to ethereals instead of punches made by ethereals deal burn damage stuck around for quite a while)).

I'd like to repeat, though, that I'd much prefer it that no race be removed to "make room" for gehennites at all- doing so would be completely unnecessary and would make no sense.

**Comment by Militaires** [comment]
oranges is the one who chooses that not me

**Comment by ATH1909** [comment]
> oranges is the one who chooses that not me

Did you not choose for moths to be the removed race (and not some other race?)?

**Comment by ATH1909** [comment]
> i think all blind people are flashproof anyhow, lizards, or moths, maintainers get to choose fams not me

Them being flashproof will be an interesting "pro" to the race; I quite liked being a flashproof moth while that moth flashproofing bug was around, and a fully flashproof race (not just a race that gets a flashproof helmet) would be nice to have/play as.

**Comment by Militaires** [comment]
@ATH1909 moths are the biggest candidates for removal since lizards would be pretty hard, maybe flies because you can already become one through teleporter after roundstart, it's up to oranges either way

**Comment by datorangebottle** [comment]
> @ATH1909 moths are the biggest candidates for removal since lizards would be pretty hard, maybe flies because you can already become one through teleporter after roundstart, it's up to oranges either way

Aren't flies pretty much already gone?

**Comment by Militaires** [comment]
dont know if they're still roundstart really

**Comment by Nickkoranda** [comment]
Flies are removed from the round-start races

**Comment by PKPenguin321** [comment]
>>So is the idea to just sit in a corner with a medibot hitting yourself with IEDs or

>you could at least read the whole pr before complaining

Unless there's something I'm missing I don't see why what I said wouldn't work? It's rare but not impossible and something our players would definitely grind out

That being said I like these, echolocation looks awesome

**Comment by Militaires** [comment]
@PKPenguin321 >Unless there's something I'm missing 

on Jah there is....

**Comment by zamolxius** [comment]
so echolocation lets them see in the dark,  very cool,  but what does flashbang to to gehennite?

**Comment by Militaires** [comment]
currently, flashbangs and loud noises do double ear damage to gehennites because of their sensitive organs, i'm also gonna have a huge overlay pop up on your screen effectively scrambling your echolocation for a short time

**Comment by Militaires** [comment]
currently, flashbangs and loud noises do double ear damage to gehennites because of their sensitive organs, i'm also gonna have a huge overlay pop up on your screen effectively scrambling your echolocation for a short time

**Comment by Mickyan** [comment]
Hell yeah lets remove a perfectly fine race in favor of another gimmick that everyone will be tired of in a week

**Comment by Barhandar** [comment]
What Mickyan said, also, these gehennites don't look like they'd provide roleplay variation either unlike moths and others.

What they _would_ provide is ability to immediately go off to validhunt with the roundstart pseudo-nightvision.

**Comment by ATH1909** [comment]
> @ATH1909 moths are the biggest candidates for removal since lizards would be pretty hard, maybe flies because you can already become one through teleporter after roundstart, it's up to oranges either way

And ethereals are not on that list because...?

Also, flypeople are already gone, but oranges has said that you can't consider this race to be a replacement for flypeople instead of another race, because ???

**Comment by 4dplanner** [comment]
Echolocation mechanic cool, race lore very cringe

**Comment by habader** [comment]
Replace catgirls instead

**Comment by knacker48** [comment]
I feel like this could be an admin spawn like synths rather than a round-start race, although if you did that you'd probably want the mutant form to be a bit more likely to spice things up.

Whatever ends up happening with this I hope atleast some of this gets used in one form or another.

**Comment by Dorsisdwarf** [comment]
planner is right. Its a neat mechanic shackled to a shitty race

**Comment by nemvar** [comment]
A thing to think about would be what happens when you decap them, surgically reattach the head, and defib. Because if this uses a nodrop collar, it would get removed by that. Might be time to implement a NODEFIB flag because currently everything with a heart can be defibbed, even if it's a plasmamen or slimeperson with a retroactively implanted heart.

Another thing is that it's unclear what happens when someone uses prides/magic mirror to turn into this race since they would be missing the nodrop item.

**Comment by zxaber** [comment]
What if we compromise and give moths echolocation?

**Comment by nero1024** [comment]
Why not make the sprites into some kind of batperson-like creature?

**Comment by Shadowflame909** [comment]
This entire race does remind me of vampires in a sense.

Echolocation, A True Warped Form. Permadeath. Yup, this is the vampire lord from skyrim

**Comment by anconfuzedrock** [comment]
I mean we did remove flies, could be a replacement but if not lmao yes echomoths

**Comment by Mickyan** [comment]
I'm not going to bother repeating the same arguments that have been made but there's one thing that's been overlooked so far.

Races that are so fundamentally gameplay-changing do not belong at roundstart. They do not add flavor, they do not allow you to play a character. They railroad you into a miserable play-style that you will most definitely get tired of very quickly and then never use again.
We've already seen this happen with quirks, but those at least aren't taking away something in return.

Plasmamen almost became extinct on account of being slightly slower, what makes you think anyone is going to put up with being partially blind and constantly have to squint to see what is happening

**Comment by Qustinnus** [comment]
the entire "0.000001% chance to actually have the interesting thing" happen is super weak, I think its better to focus on the general gameplay rather than overdeveloping the edge-case. -1 from me unless you replace the huge sprite sicko mode with something more interesting

@nero1024 lmao ofcourse you'd want it to be antromorphic, begone furry.

**Comment by Hatfish** [comment]
> And ethereals are not on that list because...?
Because as usual everyone has forgotten that they exist.

**Comment by Timonkeyn** [comment]
can we give them a race specific syndicate item for 30 or 20 TC that removes the Genetic thingy


also what About magic Mirror in Wizards den

**Comment by theXDkidoflol** [comment]
So I dunno about you but this sounds like a huge snowflake race.

-Miss a minibomb barely
-woops new antag spawned and takes over the round because bad luck XDXD sorry
-have RNG bullshittery beat you

**Comment by tralezab** [comment]
it's so funny to hear talk about races because we can only work with the human shape for roundstart races (clothing fit) so your choice boils down to blank colored thing which end up being really fucking boring designs  or xpeople, which get torn apart by the community for being furry races.

In the end the community isn't necessarily wrong since there are a lot of horror stories about furry race metagangs and other kinds of degeneracy but it's a catch-22 isn't it

I love the echolocation just make something cool to put it on

**Comment by Militaires** [comment]
my """""lore""""" is just an excuse don't worry no one cares about lore, I heard your feedback, expect a different PR soon

**Comment by Militaires** [comment]
also this race can wear all human clothes except hats, doesn't fit, wait till I rework their placeholder sprite tho also I'm phone 
posting hwlp

**Comment by Militaires** [comment]
image
There is strong support, but also significant discontent, I'm going to make the next PR easier to swallow.
How I'll address the issues:
- I'll attempt to get it in without removing moths.
- I'll address some concerns about the sprite being too muscular and edgy (muh bubblegum), lore is some bullshit I whipped up stop crying about that though
- I'll get the non-placeholder sprites in
I still want to try to make the gene-liberated form work, I have some ideas I'm putting into action.

Code-wise, the species excluding the gehennite carbon mob is finished.

**Comment by zxaber** [comment]
My recommendations for this would be;

- Make it a proper alien race of some sort. Lizards and Moths and Plasmamen are all actual aliens (despite the first two resembling earth creatures).
- Remove the superform from being a random dice roll, and make it a race-only genetic possibility much like the lizard firebreath. You could keep the ghost swapping part, or perhaps make it a forced split personality, so that it constantly bounces between tasks and really strikes home mental instability.
- Remove most of the unrelated downsides. Their real downside is lack of color and (probably) trouble differentiating between similar-shaped mobs. Their upside is a worse xray, and presumably immunity to flashes. These alone place them about on par with most existing races (Lizards' downside is something about less cold resistance and upside is their group language, Moths have a downside of lower flash resistance and an upside of free movement when gravity is gone, and so on). The extra damage taken and the uncloneability aren't needed to balance them out, and will only serve to make fewer players consider them viable.

Also, yeah, don't remove one of the more popular races to stuff this in.

**Comment by Militaires** [comment]
echolocation doesnt just inconvenience you with color, it means you're blind 50% of the time because of the cooldown, and you can't see things moving, it's a single image, so someone can EASILY fuck your shit up during the time between your "clicks" currently it's an action button but maybe i should make it automatically "pulse", also this means guns are damn useless in your hands.

i'll remove the unrelated downsides.

**Comment by PKPenguin321** [comment]
>Also, yeah, don't remove one of the more popular races to stuff this in.

Are moths popular? Aren't there like, 2 people who play them?

>i think having it locked behind genetics is too laughably easy

Could crank the unlock difficulty to max so that the gene would take forever to figure out

**Comment by bgobandit** [comment]
this feels more naturally like a late-game R&D project, even the lore is basically that

**Comment by 123chess456** [comment]
gehennite

**Comment by Shadowflame909** [comment]
I bet this race is infact a beheaded pillarmen.

I HAVE FORSAKEN MY MORTALITY JOJO!

**Comment by PKPenguin321** [comment]
Epic reference dude!

**Comment by Borisvanmemes** [comment]
Could split the gene-liberated form off into another PR if this gets merged.

**Comment by Militaires** [comment]
-Bgo
You're right, I can lock this behind late-game RND + Illegal tech levels

-Brois
You're also right, I might have to split that off into another PR, I'll probably do that.

**Comment by ATH1909** [comment]
> > Also, yeah, don't remove one of the more popular races to stuff this in.
> 
> Are moths popular? Aren't there like, 2 people who play them?
> 
> > i think having it locked behind genetics is too laughably easy
> 
> Could crank the unlock difficulty to max so that the gene would take forever to figure out. Pretty sure that's a one line change

If you're cranking up the difficulty of the mutation to maximum (turning almost every letter into an X, forcing the geneticist to use the "joker" ability for almost every letter pairing?), maybe you should also make it guaranteed to be in every gehennite (unlike firebreath in lizards and dwarfism in all other races, which are merely in the potential dormant mutation pool for those races)?

I like the split personality idea and the idea of locking the way to let gehennites ascend behind illegal tech, though. You might want to not give the ascended form of gehennites a bunch of special abilities then, especially since there are ways to separate the two personalities of someone with a split personality and put them into two different bodies (maybe make the ascended gehennite return to normal upon dying?).

**Comment by Militaires** [comment]
These are all good ideas, I'll carry them out when I open the PR for ascended gehennites, which will be alone.

**Comment by pizzatiger** [comment]
What if the heads could unlock the collars by having three heads swipe at the same time during a red alert? The hulked out forms would be an antag for sure due to the permanent rage but it would be the type of thing that you unleash in hope that it also takes out your enemies

**Comment by Militaires** [comment]
i was also thinking something like that but it's also still too easy

**Comment by Fox-McCloud** [comment]
the echo-location thing is pretty neat.

Kinda dumb that moths have to go, though, for this.

**Comment by PKPenguin321** [comment]
lamp epic :D

**Comment by FantasticFwoosh** [comment]
Reminds me of my Geordie La-Forge [molemen idea i put up years ago](https://tgstation13.org/phpBB/viewtopic.php?f=9&t=6282) that somehow morphed into skaven/ratmen discussion. Remove their inverse sunglass eye visors for echolocation mode.

Nice work OP anyway, very cool visual effects.

**Comment by Shapsy** [comment]
Interesting and coherent lore, interesting looking core echolocation mechanic, I think it looks really cool, and definitely is more interesting than just "humans but moth". 

I question whether or not such an incredibly low chance for gene suppression to break defeats the point of the mechanic in the first place, though I guess it'd be better than it happening every other round.

Rather than having them be completely worse than humans in every way, it might be neat to play off the lore of their enhanced strength (even while mostly Gene suppressed) and do something like allow them to carry bodies or crates in hand after a delay of picking them up. That would help explain why NT decided to cut the poor things loose on their stations: they may be blind and shifty, but it's not like you need to have 20/20 vision to lug a big heavy crate around anyway

**Comment by Tlaltecuhtli** [comment]
how about an emag function ;)

**Comment by anconfuzedrock** [comment]
Shaps over here with the good idea crate man crate man crate man
The black and white outlines might get super dull but I dunno how color could make sense

**Comment by Cyprex** [comment]
>I can't add a new species without removing an old one
Yeah, what the fuck is this policy? That just seems like a dumb policy to get people to accept removals more easily. Also, Ethereals seem a lot rarer than moths to me.

**Comment by Razharas** [comment]
With this you can RP as this guy
https://www.youtube.com/watch?v=ob-P2a6Mrjs

**Comment by OverlyGeneric** [comment]
I really like the sprites, with much of the appearance being Resident Evil-like, but I feel like its too much for a playable race and more suited for some sort of space ruin megafauna.

The only question I ask is when you're going to be add floating brain people? :moyai:

**Pull request closed without merging**